### ISSUE TITLE
openstack - create install-config failed to fetch dependency of "Base Domain"

### ISSUE THREAD

**Issue by dlbewley** [openshift/installer#2997]
# Version

- 4.4.0 nightly build of mac installer

```
~/Downloads/openshift-install version
/Users/dlbewley/Downloads/openshift-install v4.4.0
built from commit 96d416ee36d8660c3f2cb33863587455278f4ed2
release image quay.io/openshift-release-dev/ocp-release-nightly@sha256:e4c521b17c773c4c0b1bd5816dc5f1660df16da8a28d68dc08394b879e8c6d91
```

# Platform:

- OpenStack OSP 13

# What happened?

I am retrying OCP to OSP install after SSC support seemingly added. I can no longer even generate an `install-config.yaml`. This was working sometime back on 4.3 nightly. 
There have been no changes to `~/.config/openstack/*`.

```console
╰─ ./install-prep.sh nightly
Initializing an installation configuration for 'nightly' environment on 'osp' infrastructure provider.
Storing installation configs in '/Users/dlbewley/src/ansible/playbook-openshift/installations/osp-nightly'.

Press enter to continue:

DEBUG OpenShift Installer v4.4.0
DEBUG Built from commit 96d416ee36d8660c3f2cb33863587455278f4ed2
DEBUG Fetching Install Config...
DEBUG Loading Install Config...
DEBUG   Loading SSH Key...
DEBUG   Loading Base Domain...
DEBUG     Loading Platform...
DEBUG   Loading Cluster Name...
DEBUG     Loading Base Domain...
DEBUG     Loading Platform...
DEBUG   Loading Pull Secret...
DEBUG   Loading Platform...
DEBUG   Fetching SSH Key...
DEBUG   Generating SSH Key...
? SSH Public Key /Users/dlbewley/.ssh/id_rsa.pub
DEBUG   Fetching Base Domain...
DEBUG     Fetching Platform...
DEBUG     Generating Platform...
? Platform openstack
INFO Credentials loaded from file "/Users/dlbewley/.config/openstack/clouds.yaml"
? Cloud shiftstack
INFO Credentials loaded from file "/Users/dlbewley/.config/openstack/clouds-public.yaml"
FATAL failed to fetch Install Config: failed to fetch dependency of "Install Config": failed to fetch dependency of "Base Domain": failed to generate asset "Platform": cloud overcloud does not exist in clouds-public.yaml
```

- ./osp-nightly/.openshift_install.log

```
time="2020-01-27T15:27:47-08:00" level=debug msg="OpenShift Installer v4.4.0"
time="2020-01-27T15:27:47-08:00" level=debug msg="Built from commit 96d416ee36d8660c3f2cb33863587455278f4ed2"
time="2020-01-27T15:27:47-08:00" level=debug msg="Fetching Install Config..."
time="2020-01-27T15:27:47-08:00" level=debug msg="Loading Install Config..."
time="2020-01-27T15:27:47-08:00" level=debug msg="  Loading SSH Key..."
time="2020-01-27T15:27:47-08:00" level=debug msg="  Loading Base Domain..."
time="2020-01-27T15:27:47-08:00" level=debug msg="    Loading Platform..."
time="2020-01-27T15:27:47-08:00" level=debug msg="  Loading Cluster Name..."
time="2020-01-27T15:27:47-08:00" level=debug msg="    Loading Base Domain..."
time="2020-01-27T15:27:47-08:00" level=debug msg="    Loading Platform..."
time="2020-01-27T15:27:47-08:00" level=debug msg="  Loading Pull Secret..."
time="2020-01-27T15:27:47-08:00" level=debug msg="  Loading Platform..."
time="2020-01-27T15:27:47-08:00" level=debug msg="  Fetching SSH Key..."
time="2020-01-27T15:27:47-08:00" level=debug msg="  Generating SSH Key..."
time="2020-01-27T15:27:51-08:00" level=debug msg="  Fetching Base Domain..."
time="2020-01-27T15:27:51-08:00" level=debug msg="    Fetching Platform..."
time="2020-01-27T15:27:51-08:00" level=debug msg="    Generating Platform..."
time="2020-01-27T15:27:53-08:00" level=info msg="Credentials loaded from file \"/Users/dlbewley/.config/openstack/clouds.yaml\""
time="2020-01-27T15:27:57-08:00" level=info msg="Credentials loaded from file \"/Users/dlbewley/.config/openstack/clouds-public.yaml\""
time="2020-01-27T15:27:57-08:00" level=fatal msg="failed to fetch Install Config: failed to fetch dependency of \"Install Config\": failed to fetch dependency of \"Base Domain\": failed to generate asset \"Platform\": cloud overcloud does not exist in clouds-public.yaml"
```

# What you expected to happen?

No errors and `install-config.yaml` created in `./osp-nightly` install dir.

# How to reproduce it (as minimally and precisely as possible)?

```console
# verify openstack client
$ export OS_CLOUD=shiftstack
$ openstack flavor list
...
$ ~/Downloads/openshift-installer create install-config --log-level=debug --dir=osp-nightly
```

# Anything else we need to know?

I do have a `project_name` in my clouds.yaml.

- ~/.config/openstack/clouds.yaml

```yaml
clouds:
  shiftstack:
    profile: overcloud
    auth:
      project_name: shiftstack
      user_domain_name: Default
      project_domain_name: Default
```

- ~/.config/openstack/clouds-public.yaml

```yaml
public-clouds:
  # overcloud is the production cloud for the masses
  overcloud:
    auth_type: password
    identity_api_version: '3'
    auth:
      auth_url: 'https://openstack.example.com:13000/v3'
    interface: public
```

- ~/.config/openstack/secure.yaml

```yaml
clouds:
  shiftstack:
    auth:
      username: 'shiftstack'
      password: 'password'
```

# References

- #2710 
- https://bugzilla.redhat.com/show_bug.cgi?id=1776272

**Comment by dlbewley** [comment]
FWIW, copying the previously working `install-config.yaml` results in the following error:

```
╰─ ./install-cluster.sh nightly
...

DEBUG OpenShift Installer v4.4.0
DEBUG Built from commit 96d416ee36d8660c3f2cb33863587455278f4ed2
DEBUG Fetching Terraform Variables...
DEBUG Loading Terraform Variables...
DEBUG   Loading Cluster ID...
DEBUG     Loading Install Config...
DEBUG       Loading SSH Key...
DEBUG       Loading Base Domain...
DEBUG         Loading Platform...
DEBUG       Loading Cluster Name...
DEBUG         Loading Base Domain...
DEBUG         Loading Platform...
DEBUG       Loading Pull Secret...
DEBUG       Loading Platform...
INFO Credentials loaded from file "/Users/dlbewley/.config/openstack/clouds.yaml"
INFO Credentials loaded from file "/Users/dlbewley/.config/openstack/clouds-public.yaml"
FATAL failed to fetch Terraform Variables: failed to load asset "Install Config": invalid "install-config.yaml" file: [platform.openstack.externalNetwork: Internal error: could not retrieve valid networks,
platform.openstack.computeFlavor: Internal error: could not retrieve valid flavors, platform.openstack.trunkSupport: Internal error: could not retrieve networking extension aliases, platform.openstack.octaviaSupport: Internal error: could not retrieve service catalog]
```

```
╰─ grep externalNet osp-nightly/install-config.yaml
    externalNetwork: floating
╰─ openstack network show floating -c id -f value
c95e1f09-fe50-4837-99d7-cf9b1c3195c1
```

**Comment by Fedosin** [comment]
Hi! this patch changed the behavior a little https://github.com/openshift/installer/pull/2664. Instead of reading all possible clouds, LoadCloudsYAML() now takes data only from clouds.yaml

#3000 should fix your problem

**Comment by Fedosin** [comment]
During generation of the `platform` asset we call `Platform()` function.
https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platform.go#L71
It asks to return all possible cloud names:
https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/openstack/openstack.go#L18

And this function ignores other cloud locations (clouds-public.yaml, secure.yaml)
https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/openstack/realvalidvaluesfetcher.go#L25

This behavior was fixed in #3000

**Comment by dlbewley** [comment]
Confirmed this is resolved for me in current 4.4 nightly build.
